android/src/main/java/uk/co/playerdata/reactnativemcumanager/DeviceUpdate.kt

```diff
@@ -61,50 +67,45 @@ class DeviceUpdate(
         } catch (e: IOException) {
             e.printStackTrace()
             disconnectDevice()
-            manager.unsetUpdate()
-            promise.reject(e)
+            promise?.reject(e)
         } catch (e: McuMgrException) {
             e.printStackTrace()
             disconnectDevice()
-            manager.unsetUpdate()
-            promise.reject(e)
+            promise?.reject(e)
         }
     }
 
     override fun onUpgradeStarted(controller: FirmwareUpgradeController) {}
 
     override fun onStateChanged(prevState: FirmwareUpgradeManager.State, newState: FirmwareUpgradeManager.State) {
         val stateMap = Arguments.createMap()
-        stateMap.putString("bleId", device.address)
+        stateMap.putString("id", id)
         stateMap.putString("state", newState.name)
         manager.updateStateCB(stateMap)
```

Comment:
N.B this should be `upgradeStateChanged` now

```suggestion
        manager.upgradeStateCB(stateMap)
```